examples/grid.cpp

```diff
@@ -305,42 +397,94 @@ DrmRenderer::Flipper::~Flipper()
 
 bool DrmRenderer::Flipper::init()
 {
+    if (!m_fpsPlane->init())
+        return false;
     if (pthread_create(&m_thread, NULL, start, this)) {
         ERROR("create thread failed");
         return false;
     }
     return true;
 }
 
+bool DrmRenderer::Flipper::waitingRenderTime()
+{
+    bool late = false;
+    if (!m_fps)
+        return late;
+    timeval current;
+    if (m_firstFrame) {
+        m_firstFrame = false;
+        gettimeofday(&m_nextTime, NULL);
+        m_duration.tv_sec = 0;
+        m_duration.tv_usec = 1000 * 1000 / m_fps;
+    }
+    else {
+        bool neverSleep = true;
+        do {
+            gettimeofday(&current, NULL);
+            timeval refresh;
+            refresh.tv_sec = 0;
+            refresh.tv_usec = 1000 * 1000 / 60;
+            timeval nextRefresh;
+            timeradd(&current, &refresh, &nextRefresh);
+
+            if (timercmp(&nextRefresh, &m_nextTime, < )) {
+                usleep(refresh.tv_usec);
```

Comment:
Hardware decoder is fast, when we are here, it usually early than it should be, so we sleep here to wait render time.  We want know the render fps so we cannot drop frame. In addition, we do not need keep one frame display again since it already a front buffer on fb